### PR TITLE
added functionality for adding and saving markers with info

### PR DESCRIPTION
**Pull request by emilytran99**
- made createMarkerForEdit function for adding marker with text box to map

- createMarkerForDisplay function for saving info to map corresponding to marker at clicked coordinates with info from createMarkerForEdit

- made buildInfoWindowInput that builds the div to display the corresponding info window to marker

### CODE REVIEW

File: src/main/webapp/js/user-map-loader.js [hunk 1]
```diff
@@ -11,9 +11,75 @@ function createMap(){
...
+  
+  //opens info window corresponding to marker
+  infoWindow.open(map, editMarker);
+}
+
+//creates marker that stores info from user
+function createMarkerForDisplay(lat, lng, content){
+  //creates marker
+  const marker = new google.maps.Marker({
+    position: {lat: lat, lng: lng},
```
Comment (joverde) [inline src/main/webapp/js/user-map-loader.js:50]: I saw that this is how they did it in the guide, but I was just wondering if you thought it was confusing to have repeated values (lat = lat, lng = lng)? Just wanted to ask about it.

File: src/main/webapp/js/user-map-loader.js [hunk 2]
```diff
@@ -11,9 +11,75 @@ function createMap(){
...
-    });  
+    createMarkerForEdit(event.latLng.lat(), event.latLng.lng());
+  });
+}
+
+//creates marker at given lat and lng coordinates with 
+  //corresponding info window
+function createMarkerForEdit(lat, lng){
+  //closes previously opened marker if exists
+  if(editMarker){
```
Comment (joverde) [inline src/main/webapp/js/user-map-loader.js:22]: Saw this chunk a few times, might it be a good idea to separate into a helper function?

File: src/main/webapp/js/user-map-loader.js (hunk 1 above)
Comment (emilytran99) [reply src/main/webapp/js/user-map-loader.js:50]: I see your point but I felt like lat and lng are just the given name for working with Maps markers and it just made sense to continue lat and lng as parameter names bc it had also been the name I'd been working with throughout the code.

File: src/main/webapp/js/user-map-loader.js (hunk 2 above)
Comment (emilytran99) [reply src/main/webapp/js/user-map-loader.js:22]: which part? @joverde

Comment (joverde) [reply src/main/webapp/js/user-map-loader.js:22]: The part where you are deleting the marker, its right below this, starting with if(editMarker) { ... }

Comment (emilytran99) [reply src/main/webapp/js/user-map-loader.js:22]: setMap is the only line that deletes the marker I think and it's already a separate function

**Review by joverde — APPROVED** [review]
Looks good!

**Pull request merged**